Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AuthMethodManager.java

```diff
@@ -132,23 +137,26 @@ private synchronized AzureManager getAzureManager(final AuthMethod authMethod) t
         AzureManager localAzureManagerRef = azureManager;
 
         if (localAzureManagerRef == null) {
-            final String accountEmail = authMethodDetails.getAccountEmail();
             switch (authMethod) {
                 case AD:
-                    if (StringUtils.isNullOrEmpty(accountEmail) ||
-                            (!AdAuthManager.getInstance().isSignedIn() &&
-                                    !AdAuthManager.getInstance().tryRestoreSignIn(authMethodDetails))) {
+                    if (isBlank(authMethodDetails.getAccountEmail())) {
```

Comment:
Shall we merge the AD and DC parts, seems some of the codes are duplicated

Reply:
fix it.